--- a/test/h5cpp_tests/node_tests/node_iterator_test.py
+++ b/test/h5cpp_tests/node_tests/node_iterator_test.py
@@ -123,8 +123,15 @@
                  h5cpp.Path("/entry/instrument/detector_2"),
                  h5cpp.Path("/entry/instrument/detector_2/data")
                  ]
-        self.assertListEqual(
-            node_paths, paths, "Path comparison: {}".format(node_paths))
+        self.assertEqual(
+            len(paths), len(node_paths),
+            "Number of links comparison: {} == {}".format(len(paths),
+                                                          len(node_paths)))
+
+        for path in paths:
+            self.assertTrue(
+                path in node_paths,
+                "Path comparison: {} in {}".format(path, node_paths))

     def test_exists(self):

--- a/test/h5cpp_tests/node_tests/link_iteration_test.py
+++ b/test/h5cpp_tests/node_tests/link_iteration_test.py
@@ -88,8 +88,9 @@

         links = [link.path.name for link in entry.links]

-        self.assertListEqual(links, ["data", "instrument", "sample"],
-                             "Comparison of link names {}".format(links))
+        self.assertSetEqual(set(links),
+                            set(["data", "instrument", "sample"]),
+                            "Comparison of link names {}".format(links))

     def test_recursive_iteration(self):

@@ -104,5 +105,5 @@
                     "detector_module_2",
                     "detector_module_3",
                     "sample"]
-        self.assertListEqual(links, refnames,
-                             "Comparison of link names {}".format(links))
+        self.assertSetEqual(set(links), set(refnames),
+                            "Comparison of link names {}".format(links))
